modules/masterbar/masterbar.php

```diff
@@ -68,6 +118,50 @@ function __construct() {
 		add_action( 'wp_logout', array( $this, 'maybe_logout_user_from_wpcom' ) );
 	}
 
+	public function tracks_and_redirect_add_route() {
+		add_rewrite_rule( 'jetpack-track-and-bounce.php\?tracks_and_bounce=(\w*)', 'index.php?tracks_and_bounce=$matches[1]', 'top' );
+	}
+
+	public function tracks_and_redirect_add_query_vars( $query_vars ) {
+		$query_vars[] = 'tracks_and_bounce';
+		$query_vars[] = 'tracks_and_bounce_nonce';
+		return $query_vars;
+	}
+
+	public function tracks_and_redirect_parse_request( &$wp ) {
```

Comment:
Do we need to pass by reference here?

Reply:
Correct again! Updated on a36123157b1f1350f06c11dc27d73f714c68a981.